src/utils/expenses.js

```diff
@@ -0,0 +1,59 @@
+const { Op } = require('sequelize');
+
+const { getValidString } = require('./validation');
+const { getErrorWithStatus } = require('./getError');
+
+const isExpenseValid = (expense) => {
+  const { title, category, spentAt, note } = expense;
+
+  const textFields = {
+    title,
+    category,
+    spentAt,
+    note,
+  };
+
+  Object.entries(textFields).forEach(([key, value]) => {
+    const field = key.toString();
+
+    if ((field === 'note' || field === 'category') && !value) {
+      return;
+    }
+
+    getValidString(value, field);
+  });
+
+  if (typeof expense.amount !== 'number') {
+    throw getErrorWithStatus(400, `Type of amount must be string`);
```

Comment:
The error message states that the type of amount must be 'string', but the condition checks if it is not a 'number'. This seems contradictory. Ensure the error message accurately reflects the expected type.